### PR TITLE
Unify k8s api endpoints

### LINKED ISSUE #25: Remove hacks caused by https://github.com/kubernetes/kubernetes/issues/23831
We need to get rid of all `k8s1` and `k8s2` occurences. The bug was fixed here: https://github.com/kubernetes/kubernetes/pull/28414

### PR DESCRIPTION
**Pull request by nebril**
fixes #25

---

This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-appcontroller/29)

### CODE REVIEW

**Comment by nhlfr** [comment]
LGTM

File: client/resdef.go [hunk 1]
```diff
@@ -55,14 +55,7 @@ type resourceDefinitions struct {
...
 func newResourceDefinitions(c restclient.Config) (*resourceDefinitions, error) {
-	c.APIPath = "/apis"
-	c.ContentConfig = restclient.ContentConfig{
-		GroupVersion: &unversioned.GroupVersion{
-			Group:   "appcontroller.k8s2",
-			Version: "v1alpha1",
-		},
-		NegotiatedSerializer: api.Codecs,
-	}
+	thirdPartyResourceConfigInit(&c)
```
Comment (gluke77) [inline client/resdef.go]: this and below duplicates the same in dependencies.go. Could it be refactored somehow?

**Comment by gluke77** [comment]
LGTM. It's a bit weird that the client contains deps and resdefs, both referring to the same restclient, but i can not see better way,at least right now.

**Comment by nebril** [comment]
This should be merged after https://github.com/kubernetes/kubernetes/pull/28414 is released

**Comment by nebril** [comment]
The fix will be released in 1.4 afaik.

**Comment by pigmej** [comment]
https://github.com/kubernetes/kubernetes/pull/28414 is merged now so it will be included in 1.4 release.

**Comment by pigmej** [comment]
@nebril Can we add there a check for a k8s version inside AC? Then it would crash with useful message.

**Comment by nebril** [comment]
@pigmej done

**Comment by jellonek** [comment]
:lgtm:

---

Reviewed 21 of 21 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-appcontroller/29#-:-KVdXydFuP06-6xhy8mX:bnfp4nl)*

**Pull request merged**